### PR TITLE
Feature/tclune/generic refactor jan05 2022

### PR DESCRIPTION
**Pull request by tclune**
## Description

Refactoring MAPL_GenericInitialize() for simplicity.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Trivial change (affects only documentation or cleanup)

## Checklist:

- [x] I have tested this change with a run of GEOSgcm (if non-trivial)
- [x] I have added one of the required labels (0 diff, 0 diff trivial, 0 diff structural, non 0-diff)
- [x] I have updated the CHANGELOG.md accordingly following the style of [Keep a Changelog](https://keepachangelog.com/en/1.0.0/#how)

### CODE REVIEW

File: generic/MAPL_Generic.F90 [hunk 1]
```diff
@@ -758,8 +758,8 @@ end subroutine MAPL_GenericSetServices
    ! !IROUTINE: MAPL_GenericInitialize -- Initializes the component and its children
 
    ! !INTERFACE:
-   recursive subroutine MAPL_GenericInitialize ( GC, IMPORT, EXPORT, CLOCK, RC )
-
+   recursive subroutine MAPL_GenericInitialize ( GC, import, EXPORT, CLOCK, RC )
```
Comment (mathomp4) [inline generic/MAPL_Generic.F90:761]: Should all the arguments be lowercase if you change one?

Comment (tclune) [reply generic/MAPL_Generic.F90:761]: Not - sure how that happened - was not on my agenda today.   Probably best to just leave the PR as changed.   consistent capitalization is already out between my style and that of others.  Someday ...

**Comment by mathomp4** [comment]
Blocking until #1275 and #1276 are fixed

**Comment by mathomp4** [comment]
Actually, this might fix #1275 and #1276 as it has the `recursive` in the needed routine. I will test.

File: generic/MAPL_Generic.F90 (hunk 1 above)
Comment (mathomp4) [reply generic/MAPL_Generic.F90:761]: okay. Resolving.

**Comment by mathomp4** [comment]
Welp, this fixes the two issues it seems.

Closes #1275 

Closes #1276

**Review by mathomp4 — APPROVED** [review]
Approve

**Pull request merged**